plugins/generic/recommendBySimilarity/RecommendBySimilarityPlugin.php

```diff
@@ -8,104 +8,130 @@
  * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
  *
  * @class RecommendBySimilarityPlugin
+ *
  * @brief Plugin to recommend similar articles.
  */
 
 namespace APP\plugins\generic\recommendBySimilarity;
 
 use APP\core\Application;
+use APP\facades\Repo;
+use APP\handler\Handler;
 use APP\search\ArticleSearch;
+use APP\submission\Collector;
+use APP\submission\Submission;
+use APP\template\TemplateManager;
 use PKP\plugins\GenericPlugin;
 use PKP\plugins\Hook;
 
-define('RECOMMEND_BY_SIMILARITY_PLUGIN_COUNT', 10);
-
 class RecommendBySimilarityPlugin extends GenericPlugin
 {
-    //
-    // Implement template methods from Plugin.
-    //
+    private const DEFAULT_RECOMMENDATION_COUNT = 10;
+
     /**
      * @copydoc Plugin::register()
      *
      * @param null|mixed $mainContextId
      */
-    public function register($category, $path, $mainContextId = null)
+    public function register($category, $path, $mainContextId = null): bool
     {
-        $success = parent::register($category, $path, $mainContextId);
-        if (Application::isUnderMaintenance()) {
-            return $success;
+        if (!parent::register($category, $path, $mainContextId)) {
+            return false;
         }
 
-        if ($success && $this->getEnabled($mainContextId)) {
-            Hook::add('Templates::Article::Footer::PageFooter', [$this, 'callbackTemplateArticlePageFooter']);
+        if (!Application::isUnderMaintenance() && $this->getEnabled($mainContextId)) {
+            Hook::add('Templates::Article::Footer::PageFooter', function (string $hookName, array $params): bool {
+                $output = & $params[2];
+                $output .= $this->buildTemplate();
+                return Hook::CONTINUE;
+            });
         }
-        return $success;
+        return true;
     }
 
     /**
-     * @see Plugin::getDisplayName()
+     * Builds the template with the recommended submissions or null if the linked submission has no keywords
+     *
+     * @see templates/article/footer.tpl
      */
-    public function getDisplayName()
+    private function buildTemplate(): ?string
     {
-        return __('plugins.generic.recommendBySimilarity.displayName');
+        $templateManager = TemplateManager::getManager();
+        $submissionId = $templateManager->getTemplateVars('article')->getId();
+
+        // If there's no keywords, quit
+        if (!strlen($searchPhrase = implode(' ', (new ArticleSearch())->getSimilarityTerms($submissionId)))) {
+            return null;
+        }
+
+        $request = Application::get()->getRequest();
+        $router = $request->getRouter();
+        $context = $router->getContext($request);
+
+        $rangeInfo = Handler::getRangeInfo($request, 'articlesBySimilarity');
+        $rangeInfo->setCount(static::DEFAULT_RECOMMENDATION_COUNT);
```

Comment:
The `$rangeInfo` has access to the page and the count, but it doesn't compute the offset, is this expected or a bug?

Reply:
Indeed